lib/utils/colors.js

```diff
@@ -2,132 +2,120 @@ const Settings = {
   enabled: process.env.COLORS !== '0'
 };
 
-function Background() {
-  return this;
-}
+const foregroundColors = {
+  black: '0;30',
+  dark_gray: '1;30',
+  blue: '0;34',
+  light_blue: '1;34',
+  green: '0;32',
+  light_green: '1;32',
+  cyan: '0;36',
+  light_cyan: '1;36',
+  red: '0;31',
+  light_red: '1;31',
+  purple: '0;35',
+  light_purple: '1;35',
+  brown: '0;33',
+  yellow: '1;33',
+  light_gray: '0;37',
+  white: '1;37',
+  stack_trace: '0;90'
+};
 
-function ConsoleColor() {
-  this.background = new Background();
-
-  const foregroundColors = {
-    black: '0;30',
-    dark_gray: '1;30',
-    blue: '0;34',
-    light_blue: '1;34',
-    green: '0;32',
-    light_green: '1;32',
-    cyan: '0;36',
-    light_cyan: '1;36',
-    red: '0;31',
-    light_red: '1;31',
-    purple: '0;35',
-    light_purple: '1;35',
-    brown: '0;33',
-    yellow: '1;33',
-    light_gray: '0;37',
-    white: '1;37',
-    stack_trace: '0;90'
-  };
-
-  const backgroundColors = {
-    black: '40',
-    red: '41',
-    green: '42',
-    yellow: '43',
-    blue: '44',
-    magenta: '45',
-    cyan: '46',
-    light_gray: '47'
-  };
-
-  Object.keys(foregroundColors).forEach(k => {
-    ConsoleColor.prototype[k.toLowerCase()] = (text, background) => {
-      if (!Settings.enabled) {
-        return text;
-      }
+const backgroundColors = {
+  black: '40',
+  red: '41',
+  green: '42',
+  yellow: '43',
+  blue: '44',
+  magenta: '45',
+  cyan: '46',
+  light_gray: '47'
+};
+
+class Background {}
 
-      let string = `\u{1b}[${foregroundColors[k.toLowerCase()]}m`;
-      if (background !== undefined) {
-        string += background();
-      }
+class ConsoleColor {
+  constructor() {
+    this.background = new Background();
 
-      string += `${text}\u{1b}[0m`;
+    Object.keys(foregroundColors).forEach(k => {
+      ConsoleColor.prototype[k.toLowerCase()] = (text, background) => {
```

Comment:
the 'this' keyword refers to the object instance
```suggestion
      this[k.toLowerCase()] = (text, background) => {
```

Reply:
The suggested change would add the properties on a particular class instance and not for all the instances that exist, but the goal here is to change the properties for all class instances, hence adding the properties on the class prototype.

For ex., we want the following to work:
```js
class Background {}

const background = new Background();

Background.prototype.hello = 'hi';
const background2 = new Background();

console.log(background2.hello);  // emits `'hi'`.
console.log(background.hello);  // should also emit `'hi'`.
```

Reply:
I am trying to understand the reason behind modifying class properties after initialization. In the provided code, all properties are added during the first initialization of the class.